packages/astro/src/runtime/server/render/page.ts

```diff
@@ -19,6 +20,29 @@ function nonAstroPageNeedsHeadInjection(pageComponent: NonAstroPageComponent): b
 	return needsHeadRenderingSymbol in pageComponent && !!pageComponent[needsHeadRenderingSymbol];
 }
 
+async function iterableToHTMLBytes(
+	result: SSRResult,
+	iterable: ComponentIterable,
+	onDocTypeInjection?: (parts: HTMLParts) => Promise<void>
+): Promise<Uint8Array> {
+	const parts = new HTMLParts();
+	let i = 0;
+	for await (const chunk of iterable) {
+		if (isHTMLString(chunk)) {
+			if (i === 0) {
```

Comment:
I cant see where `i` is being incremented within the for loop, it reads *forgive me, as a unnecessary conditional check, I mean its perhaps somewhere but I just cant see it from here,